libraries/ESP8266WebServer/examples/FSBrowser/readme.md

```diff
@@ -0,0 +1,121 @@
+- ? Add a visible root node "/" (with no delete option) + add the FS type next to it, like <i>LittleFS</i> 
+- ? move "Mkdir" and "MkFile" to context menu, with prompt like for Rename/Move
+- ? implement drag/drop for move + make "rename" only a local rename operation (no move)
+- ? Optionally present SPIFFS as a hierarchical FS too
+- ? Optionally mount several filesystems at the same time (SPIFFS + SDFS or LittleFS + SDFS)
+
+## Test suite
+These tests are a checklist of operations to verify the FSBrowser behaviour.
+### On SPIFFS
+#### 8.3 filenames
+- At root : MkFile '/1.txt' / List / Edit / Download / Delete / Upload '/1.png' / View image / Delete image
+- In subdir : MkFile '/dir/2.txt' / List / Edit / Download / Delete / Upload '/dir/2.png' / View image
+- Create nested file '/a/b.txt' and delete it
+- Attempt creation of unsupported filenames
+#### Long filenames
+- At root : MkFile '/My text file 1.txt' / List / Edit / Download / Delete / Upload '/My image file 1.png' / View image / Delete image
+- In subdir : MkFile '/My Directory/My text 2.txt' / List / Edit / Download / Delete / Upload '/My Directory/My image 2.png' / View image
+- Create nested file '/My folder/My test file.txt' and delete it
+
+### On LittleFS
+#### 8.3 filenames
+- At root : MkFile '/1.txt' / List / Edit / Download / Delete / Upload '/1.png' / View image / Delete image / Mkdir '/dir'
+- In subdir : MkFile '/dir/2.txt' / List / Edit / Download / Delete / Upload '/dir/2.png' / View image / Mkdir '/dir/sub'
+- Delete root folder '/dir'
+- Create nested file '/a/b.txt' and delete file 'b.txt'
+#### Long filenames
+- At root : MkFile '/My text file 1.txt' / List / Edit / Download / Delete / Upload '/My image file 1.png' / View image / Delete image / Mkdir '/My Directory'
+- In subdir : MkFile '/My Directory/My text file 2.txt' / List / Edit / Download / Delete / Upload '/My Directory/My image file 2.png' / View image / Mkdir '/My Directory/My Subdirectory'
+- Delete root folder '/My Directory'
+- Create nested file '/My folder/My test file.txt' and delete file 'My test file.txt' 
+ 
+### On SDFS
+#### 8.3 filenames
+- At root : MkFile '/1.txt' / List / Edit / Download / Delete / Upload '/1.png' / View image / Delete image / Mkdir '/dir'
+- In subdir : MkFile '/dir/2.txt' / List / Edit / Download / Delete / Upload '/dir/2.png' / View image / Mkdir '/dir/sub'
+- Delete root folder '/dir'
+- Create nested file '/a/b.txt' and delete file 'b.txt', then delete '/a'
+#### Long filenames
+- At root : MkFile '/My text file 1.txt' / List / Edit / Download / Delete / Upload '/My image file 1.png' / View image / Delete image / Mkdir '/My Directory'
+- In subdir : MkFile '/My Directory/My text file 2.txt' / List / Edit / Download / Delete / Upload '/My Directory/My image file 2.png' / View image / Mkdir '/My Directory/My Subdirectory'
+- Delete root folder '/My Directory'
+- Create nested file '/My folder/My test file.txt' and delete file 'My test file.txt' 
+
+## Credits
```

Comment:
Maybe add something like:
Original version of FSBrowser written by me-no-dev, contributions over time by various contributors